frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-topology/ApprovalTaskNode.tsx

```diff
@@ -0,0 +1,222 @@
+  const taskStatusColor = status
+    ? getApprovalStatusColor(status).pftoken.value
+    : getApprovalStatusColor(ApprovalStatus.Idle).pftoken.value;
+
+  const [hover, hoverRef] = useHover();
+  const truncatedVisualName = React.useMemo(
+    () => truncateMiddle(visualName, { length: showStatusState ? 11 : 14, truncateEnd: true }),
+    [visualName, showStatusState],
+  );
+
+  const renderVisualName = (
+    <text
+      ref={nameRef}
+      x={showStatusState ? 30 : width / 2}
+      y={height / 2 + 1}
+      className={cx('odc-pipeline-vis-task-text', {
+        'is-text-center': !pipelineRunName,
+        'is-linked': enableLogLink,
+      })}
+    >
+      {truncatedVisualName}
+    </text>
+  );
+
+  let taskPill = (
+    <g ref={hoverRef}>
+      <SvgDropShadowFilter dy={1} id={FILTER_ID} />
+      <rect
+        filter={hover ? createSvgIdUrl(FILTER_ID) : ''}
+        width={width}
+        height={height}
+        rx={5}
+        className={cx('odc-pipeline-vis-task', {
+          'is-selected': !!pipelineRunName && hover,
+          'is-linked': !!pipelineRunName && enableLogLink,
+        })}
+        style={{
+          stroke: pipelineRunName
+            ? status
+              ? getApprovalStatusColor(status).pftoken.value
+              : getApprovalStatusColor(ApprovalStatus.Idle).pftoken.value
+            : '',
+        }}
+      />
+      {visualName !== truncatedVisualName && disableVisualizationTooltip ? (
+        <Tooltip triggerRef={nameRef} content={visualName}>
+          {renderVisualName}
+        </Tooltip>
+      ) : (
+        renderVisualName
+      )}
+
+      {showStatusState && (
+        <>
+          <svg
+            width={30}
+            height={30}
+            viewBox="-10 -7 30 30"
+            style={{
+              color: taskStatusColor,
+            }}
+          >
+            <ApprovalStatusIcon status={status} />
+          </svg>
+        </>
+      )}
+    </g>
+  );
+
+  if (!disableVisualizationTooltip) {
+    taskPill = (
+      <>
+        <Tooltip
+          triggerRef={pillRef}
+          position="bottom"
+          enableFlip={false}
+          content={t('pipelines-plugin~Approval Task')}
+        >
+          <g ref={pillRef}>{taskPill}</g>
+        </Tooltip>
+      </>
+    );
+  }
+  return (
+    <g className={cx('odc-pipeline-topology__task-node', { 'is-link': enableLogLink })}>
+      {enableLogLink ? <Link to={path}>{taskPill}</Link> : taskPill}
+    </g>
+  );
+};
+
+const ApprovalTaskNode: React.FC<ApprovalTaskNodeProps> = ({ element, disableTooltip }) => {
+  const { height, width } = element.getBounds();
+
+  const { pipeline, pipelineRun, task } = element.getData();
+
+  const customTaskName = `${pipelineRun?.metadata?.name}-${task?.name}`;
+  const pipelineRunStatus = pipelineRun && pipelineRunFilterReducer(pipelineRun);
+  let customTaskStatus: string = '';
+
+  const watchedResources = {
+    customRun: {
+      kind: referenceForModel(CustomRunModelV1Beta1),
```

Comment:
avoid using `referenceForModel` as it is deprecated.
```suggestion
       groupVersionKind: getGroupVersionKindForModel(CustomRunModelV1Beta1)
```